pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java

```diff
@@ -99,7 +99,8 @@ protected void initChannel(SocketChannel ch) throws Exception {
                     ch.pipeline().addLast(TLS_HANDLER, sslCtx.newHandler(ch.alloc()));
                 }
                 ch.pipeline().addLast("frameDecoder",
-                        new LengthFieldBasedFrameDecoder(PulsarDecoder.MaxFrameSize, 0, 4, 0, 4));
+                                      new LengthFieldBasedFrameDecoder(config.getMaxMessagesSize() + 10 * 1024, 0, 4, 0,
```

Comment:
The amount of padding should also be defined in a constant (shared between all components)

Reply:
such as `pulsar-common`?

Reply:
Yes, somewhere like `Commands.MESSAGE_SIZE_FRAME_PADDING`

Reply:
I put the `FRAME_PADDING` into `InternalConfigurationData`, I think it's a configuration so it's more suit to set in `InternalConfigurationData`?